frontend/src/lib/notification.tsx

```diff
@@ -0,0 +1,34 @@
+import getUuidByString from 'uuid-by-string';
+
+export enum NOTIFICATION_TYPE {
+  DIALOG = 'DIALOG',
+  SNACKBAR = 'SNACKBAR',
+  BOTH = 'BOTH',
+}
+
+export class Notification {
+  type: NOTIFICATION_TYPE = NOTIFICATION_TYPE.DIALOG;
+  message: string = '';
+  id: string;
+  seen: boolean = false;
+  url?: string;
+  date: string = new Date().toUTCString();
+
+  constructor(type?: NOTIFICATION_TYPE, message?: string, date?: string) {
+    if (type) {
+      this.type = type;
+    }
+    if (message) {
+      this.message = message;
+    }
+    if (date) {
+      this.date = date;
+    }
+    // generate the id based on the message and the date attached to a notification
+    this.id = getUuidByString(`${date},${message}`);
```

Comment:
You could remove the extra dependency by using:
```ts
this.id = btoa(`${item.date},${item.message}`)
```

There's no need for a UUID, since no UUID properties are being used.

Reply:
TIL about btoa, Thanks I used UUID to uniquely identify each notification but btoa would do the same job here I will revert to using btoa